lib/pushmi_pullyu/fedora_config.rb

```diff
@@ -0,0 +1,46 @@
+require 'yaml'
+class PushmiPullyu::FedoraConfig
+
+  include PushmiPullyu::Logging
+
+  attr_accessor :config_file, :url, :user, :password, :base_path
+
+  def initialize
+    self.config_file = ENV['FEDORA_CONFIG_FILE'] ||
+                       File.expand_path(File.dirname(__FILE__) + '../../../config/fedora.yml')
+
+    self.url = ENV['FEDORA_URL']
+    self.user = ENV['FEDORA_USER']
+    self.password = ENV['FEDORA_PASSWORD']
+    self.base_path = ENV['FEDORA_BASEPATH']
+  end
+
+  def process_options(opts)
+    opts.on('-fc', '--fedora-config-file FILE',
+            "Location of Fedora configuration (Default: #{config_file})") do |filename|
+      config_file = filename
+    end
+
+    opts.separator ''
+
+    load_from_config_file
```

Comment:
This should probably happen after the opt parser. Since you want `pushmi_pullyu --help` or `pushmi_pullyu --version` to do their thing and exit quickly. Now it will read from this file and parse all the opts, then execute the `--help/--version`